.github/workflows/ci.yml

```diff
@@ -24,6 +24,15 @@ jobs:
           key: ${{ runner.os }}-website-${{ hashFiles('**/yarn.lock') }}
           restore-keys: |
             ${{ runner.os }}-website-
+      - name: Cache Docusaurus Build
+        uses: actions/cache@v3
+        with:
+          path: node_modules/.cache/webpack
+          key: docusaurus-${{ github.ref_name }}
```

Comment:
I think this key will cause the cache to be created once and never refreshed for each ref, which is probably fine for PRs (presumably their first build and their later builds are similar enough to likely-benefit), but... won't be good for `main`.

Maybe we could do something like https://www.pantsbuild.org/2.18/docs/using-pants/using-pants-in-ci 's coarse-grained caching with a commit or a bulk `hashFiles` in there too.

Reply:
Ok, see latest commit.

I got this from https://github.com/actions/cache/blob/main/tips-and-workarounds.md#update-a-cache.

As I understand it the `key` will always miss, and then go to the restore key. That'll essentially be the last build on this branch (or `main` in the case of a PR). The downside is it'll eat up cache quota like a hungry hungry hippo. But meh?

> A repository can have up to 10GB of caches. Once the 10GB limit is reached, older caches will be evicted based on when the cache was last accessed. Caches that are not accessed within the last week will also be evicted.

Reply:
Ah, yeah. We might want to nuke the cache when it gets too big (along the lines of the pants docs), or else each cache entry will eventually grow to be 5GB, 10GB, 20GB, ... by itself, and we'll spend more time uploading/downloading than it saves.